Cannot delete sechub job because of jackson invalid type id

When we cancel an older job - being marked as still running -  we got an exception

Interesting part of trace:

---
Caused by: com.fasterxml.jackson.databind.exc.InvalidTypeIdException: 
Missing type id when trying to resolve subtype of 
[map type; class java.util.HashMap, [simple type, class java.lang.String] -> [simple type, class java.lang.Object]]: missing type id property '@class' at [Source: (ByteArrayInputStream); line: 1, column: 2] at
 com.fasterxml.jackson.databind.exc.InvalidTypeIdException.from(InvalidTypeIdException.java:43) at
...
org.springframework.batch.core.repository.dao.**JdbcExecutionContextDao**$ExecutionContextRowMapper.mapRow(JdbcExecutionContextDao.java:325) ... 43 common frames omitted

---

Full stacktrace:

 ```
	java.lang.IllegalArgumentException:
Unable to deserialize the execution context at
 org.springframework.batch.core.repository.dao.JdbcExecutionContextDao$ExecutionContextRowMapper.mapRow(JdbcExecutionContextDao.java:328) at
 org.springframework.batch.core.repository.dao.JdbcExecutionContextDao$ExecutionContextRowMapper.mapRow(JdbcExecutionContextDao.java:312) at
 org.springframework.jdbc.core.RowMapperResultSetExtractor.extractData(RowMapperResultSetExtractor.java:94) at
 org.springframework.jdbc.core.RowMapperResultSetExtractor.extractData(RowMapperResultSetExtractor.java:61) at
 org.springframework.jdbc.core.JdbcTemplate$1.doInPreparedStatement(JdbcTemplate.java:723) at
 org.springframework.jdbc.core.JdbcTemplate.execute(JdbcTemplate.java:651) at
 org.springframework.jdbc.core.JdbcTemplate.query(JdbcTemplate.java:713) at
 org.springframework.jdbc.core.JdbcTemplate.query(JdbcTemplate.java:744) at
 org.springframework.jdbc.core.JdbcTemplate.query(JdbcTemplate.java:757) at
 org.springframework.jdbc.core.JdbcTemplate.query(JdbcTemplate.java:815) at
 org.springframework.batch.core.repository.dao.JdbcExecutionContextDao.getExecutionContext(JdbcExecutionContextDao.java:114) at
 org.springframework.batch.core.explore.support.SimpleJobExplorer.getJobExecutionDependencies(SimpleJobExplorer.java:232) at
 org.springframework.batch.core.explore.support.SimpleJobExplorer.findRunningJobExecutions(SimpleJobExplorer.java:116) at
 java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method) at
 java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(Unknown Source) at
 java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(Unknown Source) at
 java.base/java.lang.reflect.Method.invoke(Unknown Source) at
 org.springframework.aop.support.AopUtils.invokeJoinpointUsingReflection(AopUtils.java:344) at
 org.springframework.aop.framework.ReflectiveMethodInvocation.invokeJoinpoint(ReflectiveMethodInvocation.java:198) at
 org.springframework.aop.framework.ReflectiveMethodInvocation.proceed(ReflectiveMethodInvocation.java:163) at
 org.springframework.batch.core.configuration.annotation.SimpleBatchConfiguration$PassthruAdvice.invoke(SimpleBatchConfiguration.java:128) at
 org.springframework.aop.framework.ReflectiveMethodInvocation.proceed(ReflectiveMethodInvocation.java:186) at
 org.springframework.aop.framework.JdkDynamicAopProxy.invoke(JdkDynamicAopProxy.java:215) at
 com.sun.proxy.$Proxy122.findRunningJobExecutions(Unknown Source) at
 com.daimler.sechub.domain.schedule.batch.SchedulerCancelBatchJobService.stopAllRunningBatchJobsForSechubJobUUID(SchedulerCancelBatchJobService.java:60) at
 com.daimler.sechub.domain.schedule.batch.SchedulerCancelBatchJobService.stopAllRunningBatchJobsForSechubJobUUID(SchedulerCancelBatchJobService.java:39) at
 com.daimler.sechub.domain.schedule.batch.SchedulerCancelBatchJobService$$FastClassBySpringCGLIB$$1c3d2bc5.invoke(<generated>) at
 org.springframework.cglib.proxy.MethodProxy.invoke(MethodProxy.java:218) at
 org.springframework.aop.framework.CglibAopProxy$CglibMethodInvocation.invokeJoinpoint(CglibAopProxy.java:771) at
 org.springframework.aop.framework.ReflectiveMethodInvocation.proceed(ReflectiveMethodInvocation.java:163) at
 org.springframework.aop.framework.CglibAopProxy$CglibMethodInvocation.proceed(CglibAopProxy.java:749) at
 org.springframework.transaction.interceptor.TransactionAspectSupport.invokeWithinTransaction(TransactionAspectSupport.java:371) at
 org.springframework.transaction.interceptor.TransactionInterceptor.invoke(TransactionInterceptor.java:134) at
 org.springframework.aop.framework.ReflectiveMethodInvocation.proceed(ReflectiveMethodInvocation.java:186) at
 org.springframework.aop.framework.CglibAopProxy$CglibMethodInvocation.proceed(CglibAopProxy.java:749) at
 org.springframework.aop.framework.CglibAopProxy$DynamicAdvisedInterceptor.intercept(CglibAopProxy.java:691) at
 com.daimler.sechub.domain.schedule.batch.SchedulerCancelBatchJobService$$EnhancerBySpringCGLIB$$5e13b8f5.stopAllRunningBatchJobsForSechubJobUUID(<generated>) at
 com.daimler.sechub.domain.schedule.SchedulerCancelJobService.cancelJob(SchedulerCancelJobService.java:76) at
 com.daimler.sechub.domain.schedule.ScheduleMessageHandler.handleCancelJobRequested(ScheduleMessageHandler.java:125) at
 com.daimler.sechub.domain.schedule.ScheduleMessageHandler.receiveAsyncMessage(ScheduleMessageHandler.java:93) at
 com.daimler.sechub.sharedkernel.messaging.DomainMessageService$AsynchronMessageHandlerTaskAdapter.run(DomainMessageService.java:197) at
 java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(Unknown Source) at
 java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(Unknown Source) at
 java.base/java.lang.Thread.run(Unknown Source) Caused by: com.fasterxml.jackson.databind.exc.InvalidTypeIdException: Missing type id when trying to resolve subtype of [map type; class java.util.HashMap, [simple type, class java.lang.String] -> [simple type, class java.lang.Object]]: missing type id property '@class' at
 [Source: (ByteArrayInputStream); line: 1, column: 2] at
 com.fasterxml.jackson.databind.exc.InvalidTypeIdException.from(InvalidTypeIdException.java:43) at
 com.fasterxml.jackson.databind.DeserializationContext.missingTypeIdException(DeserializationContext.java:1794) at
 com.fasterxml.jackson.databind.DeserializationContext.handleMissingTypeId(DeserializationContext.java:1323) at
 com.fasterxml.jackson.databind.jsontype.impl.TypeDeserializerBase._handleMissingTypeId(TypeDeserializerBase.java:303) at
 com.fasterxml.jackson.databind.jsontype.impl.AsPropertyTypeDeserializer._deserializeTypedUsingDefaultImpl(AsPropertyTypeDeserializer.java:166) at
 com.fasterxml.jackson.databind.jsontype.impl.AsPropertyTypeDeserializer.deserializeTypedFromObject(AsPropertyTypeDeserializer.java:107) at
 com.fasterxml.jackson.databind.deser.std.MapDeserializer.deserializeWithType(MapDeserializer.java:413) at
 com.fasterxml.jackson.databind.deser.impl.TypeWrappedDeserializer.deserialize(TypeWrappedDeserializer.java:68) at
 com.fasterxml.jackson.databind.ObjectMapper._readMapAndClose(ObjectMapper.java:4526) at
 com.fasterxml.jackson.databind.ObjectMapper.readValue(ObjectMapper.java:3513) at
 org.springframework.batch.core.repository.dao.Jackson2ExecutionContextStringSerializer.deserialize(Jackson2ExecutionContextStringSerializer.java:133) at
 org.springframework.batch.core.repository.dao.Jackson2ExecutionContextStringSerializer.deserialize(Jackson2ExecutionContextStringSerializer.java:104) at
 org.springframework.batch.core.repository.dao.JdbcExecutionContextDao$ExecutionContextRowMapper.mapRow(JdbcExecutionContextDao.java:325) ... 43 common frames omitted
```

---
<sup>Albert Tregnaghi <albert.tregnaghi@daimler.com>, Daimler TSS GmbH, [imprint](https://github.com/Daimler/daimler-foss/blob/master/LEGAL_IMPRINT.md)</sup>

--- a/sechub-doc/src/docs/asciidoc/documents/operations/01_sechub_server.adoc
+++ b/sechub-doc/src/docs/asciidoc/documents/operations/01_sechub_server.adoc
@@ -3,6 +3,8 @@
 
 include::server/server_deployment.adoc[]
 
+include::04_migration_steps.adoc[]
+
 include::../shared/configuration/mapping.adoc[]
 
 include::../shared/configuration/scan_config.adoc[]

--- a/sechub-doc/src/docs/asciidoc/documents/operations/04_migration_steps.adoc
+++ b/sechub-doc/src/docs/asciidoc/documents/operations/04_migration_steps.adoc
@@ -22,6 +22,8 @@ automation was possible!
 [[section-migration-spring-batch-477]]
 ===== Post handle CVE-2020-5411 
 
+====== Update job execution context to new format
+
 Script to prepare old jobs for Spring batch CVE-2020-5411 changes
   
 With Server v0.19.0 we upgraded our Spring Boot dependencies footnote:update_0_19_0[https://github.com/Daimler/sechub/issues/472] but this led to cancelation problems on older jobs - 
@@ -30,9 +32,9 @@ this led to problems when deleting older jobs still running.footnote:problem_499
 The solution is to execute following `SQL` script on all environments:
 [source,sql]
 ----
-UPDATE batch_job_execution_context SET short_context='{"@class":"java.util.HashMap"}' WHERE short_context ='{}'<1>
+UPDATE batch_job_execution_context SET short_context='{"@class":"java.util.HashMap"}' WHERE short_context ='{}' <1>
 ----
-<1> Migrate old (and empty) Spring Batch execution context JSON to correct format: an empty `HashMap`
+<1> Migrate old execution context JSON to correct format: an empty `HashMap`
 
 
 
@@ -42,4 +44,38 @@ a flyway update script would lead to:
 ``` java
 Caused by: org.h2.jdbc.JdbcSQLSyntaxErrorException: Table "BATCH_JOB_EXECUTION_CONTEXT" not found; SQL statement:
 ```
+
+====== Update step execution context to new format
+Additionally we need to update the step context data as well
+
+[source,sql]
+----
+-- check for entries with old format
+select sc.short_context, count(*) from batch_job_execution_params p
+inner join batch_job_execution e on e.job_execution_id=p.job_execution_id
+inner join batch_job_execution_context c on c.job_execution_id= e.job_execution_id
+inner join batch_step_execution se on se.job_execution_id = e.job_execution_id
+inner join batch_step_execution_context sc on sc.step_execution_id = se.step_execution_id
+group by sc.short_context
+
+--- execute update when former query has old entries...
+update batch_step_execution_context sc set short_context='{"@class":"java.util.HashMap","batch.taskletType":"com.daimler.sechub.domain.schedule.batch.ScanExecutionTasklet","batch.stepType":"org.springframework.batch.core.step.tasklet.TaskletStep"}'
+where sc.short_context = '{"batch.taskletType":"com.daimler.sechub.domain.schedule.batch.ScanExecutionTasklet","batch.stepType":"org.springframework.batch.core.step.tasklet.TaskletStep"}' <1>
+
+-- check again for entries in old format
+select sc.short_context, count(*) from batch_job_execution_params p
+inner join batch_job_execution e on e.job_execution_id=p.job_execution_id
+inner join batch_job_execution_context c on c.job_execution_id= e.job_execution_id
+inner join batch_step_execution se on se.job_execution_id = e.job_execution_id
+inner join batch_step_execution_context sc on sc.step_execution_id = se.step_execution_id
+group by sc.short_context
+
+
+----
+<1> Migrate old execution context JSON to correct format: with `HashMap` as class
+
+====== Cancel your old running jobs
+
+When you have some still running jobs - for example because of a former database connection pool problem -
+you can now cancel the old jobs without any problems. 
 

--- a/sechub-doc/src/docs/asciidoc/documents/operations/04_migration_steps.adoc
+++ b/sechub-doc/src/docs/asciidoc/documents/operations/04_migration_steps.adoc
@@ -1,0 +1,45 @@
+// SPDX-License-Identifier: MIT
+=== Migration steps
+
+==== Database
+[WARNING]
+====
+Normally all updates in database are done by our flyway script automatically!
+
+It is very uncommon to do something manually here. Every manual changes will be described why no
+automation was possible!
+====
+
+[options="header",cols="1,3,4,4"]
+|===
+| Nr.                    | What                                                              | Why manual?                                           | Apply to version   
+//------------------------------------------------------------------------------------------------------------------------------------------------------------------------
+| {counter:migrationNr}  | <<section-migration-spring-batch-477,post handle CVE-2020-5411>>  | Table to update does not exist on new started servers | Initial run < v0.19.0-server
+
+|===
+    
+
+[[section-migration-spring-batch-477]]
+===== Post handle CVE-2020-5411 
+
+Script to prepare old jobs for Spring batch CVE-2020-5411 changes
+  
+With Server v0.19.0 we upgraded our Spring Boot dependencies footnote:update_0_19_0[https://github.com/Daimler/sechub/issues/472] but this led to cancelation problems on older jobs - 
+this led to problems when deleting older jobs still running.footnote:problem_499[https://github.com/Daimler/sechub/issues/499]
+
+The solution is to execute following `SQL` script on all environments:
+[source,sql]
+----
+UPDATE batch_job_execution_context SET short_context='{"@class":"java.util.HashMap"}' WHERE short_context ='{}'<1>
+----
+<1> Migrate old (and empty) Spring Batch execution context JSON to correct format: an empty `HashMap`
+
+
+
+Unfortunately it is not possible to automatically migrate the data by `flyway` script, because this would fail any new started server, because
+a flyway update script would lead to:
+
+``` java
+Caused by: org.h2.jdbc.JdbcSQLSyntaxErrorException: Table "BATCH_JOB_EXECUTION_CONTEXT" not found; SQL statement:
+```
+

--- a/sechub-doc/src/docs/asciidoc/documents/techdoc/01_development.adoc
+++ b/sechub-doc/src/docs/asciidoc/documents/techdoc/01_development.adoc
@@ -82,6 +82,20 @@ mkdir ~/go/src/daimler.com/
 ln -s $pathToSecHubRepo/sechub-cli/src/daimler.com/sechub ~/go/src/daimler.com/sechub
 ----
 
+==== Special developer files
+
+===== Developer property file
+Inside `~/.sechub/sechub-developer.properties` you can define some special default behaviours
+[options="header",cols="1,1,1"]
+|===
+|Key                                |Possible values/types   |Description   
+//----------------------------------------------------------------------------------------
+|sechub.integrationtest.running     |boolean                 |When `true`, every integration test is run, 
+                                                              no matter if system property is set for launch config or not   
+|===
+
+see `LocalDeveloperFileSetupSupport.java` for more details
+
 === First steps
 Here some first steps to get {sechub} running and being able to develop parts.
 

--- a/sechub-integrationtest/src/main/java/com/daimler/sechub/integrationtest/api/AsUser.java
+++ b/sechub-integrationtest/src/main/java/com/daimler/sechub/integrationtest/api/AsUser.java
@@ -763,20 +763,27 @@ public class AsUser {
         return triggerAsyncCodeScanApproveWithoutSourceUploadAndGetJobUUID(project, IntegrationTestMockMode.CODE_SCAN__CHECKMARX__GREEN__SUPERFAST,
                 "zipfile_contains_only_test1.txt.zip");
     }
+    
+    public UUID triggerAsyncWebScanGreenLongRunningAndGetJobUUID(TestProject project) {
+        UUID uuid = createWebScan(project, IntegrationTestMockMode.WEBSCAN__NETSPARKER_RESULT_GREEN__LONG_RUNNING);
+        approveJob(project, uuid);
+        return uuid;
+    }
+    
 
     public UUID triggerAsyncCodeScanWithPseudoZipUpload(TestProject project, IntegrationTestMockMode mode) {
         return triggerAsyncCodeScanApproveWithoutSourceUploadAndGetJobUUID(project, mode, "zipfile_contains_only_test1.txt.zip");
     }
 
     public UUID triggerAsyncCodeScanApproveWithoutSourceUploadAndGetJobUUID(TestProject project, IntegrationTestMockMode mode, String pathInsideResources) {
-        UUID uuid = triggerAsyncScanAndGetJobUUID(project, mode);
+        UUID uuid = triggerAsyncCodeScanAndGetJobUUID(project, mode);
         upload(project, uuid, pathInsideResources);
 
         approveJob(project, uuid);
         return uuid;
     }
 
-    public UUID triggerAsyncScanAndGetJobUUID(TestProject project, IntegrationTestMockMode runMode) {
+    private UUID triggerAsyncCodeScanAndGetJobUUID(TestProject project, IntegrationTestMockMode runMode) {
         UUID uuid = createCodeScan(project, runMode);
         assertNotNull(uuid);
         return uuid;
@@ -952,6 +959,8 @@ public class AsUser {
             return this;
         }
     }
+
+    
 
    
    

--- a/sechub-integrationtest/src/main/java/com/daimler/sechub/integrationtest/api/TestAPI.java
+++ b/sechub-integrationtest/src/main/java/com/daimler/sechub/integrationtest/api/TestAPI.java
@@ -140,7 +140,7 @@ public class TestAPI {
     }
 
     /**
-     * Waits for sechub job being done - after 5 seconds time out is reached
+     * Waits for sechub job being done (means status execution result is OK) - after 5 seconds time out is reached
      * 
      * @param project
      * @param jobUUID
@@ -149,7 +149,7 @@ public class TestAPI {
         waitForJobDone(project, jobUUID,5);
     }
     /**
-     * Waits for sechub job being done - after 5 seconds time out is reached
+     * Waits for sechub job being done (means status execution result is OK)- after 5 seconds time out is reached
      * 
      * @param project
      * @param jobUUID
@@ -184,6 +184,26 @@ public class TestAPI {
                 String status = as(getUser()).getJobStatus(project.getProjectId(), jobUUID);
                 LOG.debug(">>>>>>>>>JOB:STATUS:" + status);
                 return status.contains("STARTED");
+            }
+        });
+    }
+    
+    /**
+     * Waits for sechub job being ended - after 5 seconds time out is reached
+     * 
+     * @param project
+     * @param jobUUID
+     */
+    @SuppressWarnings("unchecked")
+    public static void waitForJobEnded(TestProject project, UUID jobUUID) {
+        LOG.debug("wait for job ended project:{}, job:{}", project.getProjectId(), jobUUID);
+
+        TestAPI.executeUntilSuccessOrTimeout(new AbstractTestExecutable(SUPER_ADMIN, 5, HttpClientErrorException.class) {
+            @Override
+            public boolean runImpl() throws Exception {
+                String status = as(getUser()).getJobStatus(project.getProjectId(), jobUUID);
+                LOG.debug(">>>>>>>>>JOB:STATUS:" + status);
+                return status.contains("ENDED");
             }
         });
     }

--- a/sechub-integrationtest/src/main/java/com/daimler/sechub/integrationtest/api/TestAPI.java
+++ b/sechub-integrationtest/src/main/java/com/daimler/sechub/integrationtest/api/TestAPI.java
@@ -189,21 +189,42 @@ public class TestAPI {
     }
     
     /**
-     * Waits for sechub job being ended - after 5 seconds time out is reached
+     * Waits for sechub job being cancele requested - after 5 seconds time out is reached
      * 
      * @param project
      * @param jobUUID
      */
     @SuppressWarnings("unchecked")
-    public static void waitForJobEnded(TestProject project, UUID jobUUID) {
-        LOG.debug("wait for job ended project:{}, job:{}", project.getProjectId(), jobUUID);
+    public static void waitForJobStatusCancelRequested(TestProject project, UUID jobUUID) {
+        LOG.debug("wait for job cancel requested project:{}, job:{}", project.getProjectId(), jobUUID);
 
         TestAPI.executeUntilSuccessOrTimeout(new AbstractTestExecutable(SUPER_ADMIN, 5, HttpClientErrorException.class) {
             @Override
             public boolean runImpl() throws Exception {
                 String status = as(getUser()).getJobStatus(project.getProjectId(), jobUUID);
                 LOG.debug(">>>>>>>>>JOB:STATUS:" + status);
-                return status.contains("ENDED");
+                return status.contains("CANCEL_REQUESTED");
+            }
+        });
+    }
+    
+
+    /**
+     * Waits for sechub job being cancele requested - after 5 seconds time out is reached
+     * 
+     * @param project
+     * @param jobUUID
+     */
+    @SuppressWarnings("unchecked")
+    public static void waitForJobResultFailed(TestProject project, UUID jobUUID) {
+        LOG.debug("wait for job failed project:{}, job:{}", project.getProjectId(), jobUUID);
+
+        TestAPI.executeUntilSuccessOrTimeout(new AbstractTestExecutable(SUPER_ADMIN, 5, HttpClientErrorException.class) {
+            @Override
+            public boolean runImpl() throws Exception {
+                String status = as(getUser()).getJobStatus(project.getProjectId(), jobUUID);
+                LOG.debug(">>>>>>>>>JOB:STATUS:" + status);
+                return status.contains("FAILED");
             }
         });
     }

--- a/sechub-integrationtest/src/test/java/com/daimler/sechub/integrationtest/scenario4/CancelJobScenario4IntTest.java
+++ b/sechub-integrationtest/src/test/java/com/daimler/sechub/integrationtest/scenario4/CancelJobScenario4IntTest.java
@@ -1,0 +1,51 @@
+// SPDX-License-Identifier: MIT
+package com.daimler.sechub.integrationtest.scenario4;
+
+import static com.daimler.sechub.integrationtest.api.TestAPI.*;
+import static com.daimler.sechub.integrationtest.scenario4.Scenario4.*;
+
+import java.util.UUID;
+
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.rules.Timeout;
+
+import com.daimler.sechub.integrationtest.api.IntegrationTestSetup;
+import com.daimler.sechub.integrationtest.api.TestProject;
+
+/**
+ * Integration tests, recording events and check happens as expected
+ * 
+ * @author Albert Tregnaghi
+ *
+ */
+public class CancelJobScenario4IntTest {
+
+    @Rule
+    public IntegrationTestSetup setup = IntegrationTestSetup.forScenario(Scenario4.class);
+
+    @Rule
+    public Timeout timeOut = Timeout.seconds(600);
+
+    TestProject project = PROJECT_1;
+
+    @Test
+    /**
+     * We start a long running job and start a cancel operation here
+     */
+    public void cancel_a_long_running_job() {
+        /* @formatter:off */
+        /* prepare */
+        UUID sechubJobUUD = as(USER_1).triggerAsyncWebScanGreenLongRunningAndGetJobUUID(project);
+        waitForJobRunning(project, sechubJobUUD);
+        
+        /* execute */
+        as(SUPER_ADMIN).cancelJob(sechubJobUUD);
+        
+        /* test */
+        waitForJobEnded(project, sechubJobUUD);
+        
+        /* @formatter:on */
+    }
+    
+}

--- a/sechub-integrationtest/src/test/java/com/daimler/sechub/integrationtest/scenario4/CancelJobScenario4IntTest.java
+++ b/sechub-integrationtest/src/test/java/com/daimler/sechub/integrationtest/scenario4/CancelJobScenario4IntTest.java
@@ -14,7 +14,7 @@ import com.daimler.sechub.integrationtest.api.IntegrationTestSetup;
 import com.daimler.sechub.integrationtest.api.TestProject;
 
 /**
- * Integration tests, recording events and check happens as expected
+ * Integration tests to check cancel operations works
  * 
  * @author Albert Tregnaghi
  *
@@ -43,7 +43,9 @@ public class CancelJobScenario4IntTest {
         as(SUPER_ADMIN).cancelJob(sechubJobUUD);
         
         /* test */
-        waitForJobEnded(project, sechubJobUUD);
+        waitForJobStatusCancelRequested(project, sechubJobUUD);
+        waitForJobResultFailed(project, sechubJobUUD);
+        
         
         /* @formatter:on */
     }

--- a/sechub-integrationtest/src/test/java/com/daimler/sechub/integrationtest/scenario4/CancelJobScenario4IntTest.java
+++ b/sechub-integrationtest/src/test/java/com/daimler/sechub/integrationtest/scenario4/CancelJobScenario4IntTest.java
@@ -10,6 +10,7 @@ import org.junit.Rule;
 import org.junit.Test;
 import org.junit.rules.Timeout;
 
+import com.daimler.sechub.integrationtest.api.IntegrationTestMockMode;
 import com.daimler.sechub.integrationtest.api.IntegrationTestSetup;
 import com.daimler.sechub.integrationtest.api.TestProject;
 
@@ -33,10 +34,31 @@ public class CancelJobScenario4IntTest {
     /**
      * We start a long running job and start a cancel operation here
      */
-    public void cancel_a_long_running_job() {
+    public void cancel_a_long_running_webscan_job() {
         /* @formatter:off */
         /* prepare */
         UUID sechubJobUUD = as(USER_1).triggerAsyncWebScanGreenLongRunningAndGetJobUUID(project);
+        waitForJobRunning(project, sechubJobUUD);
+        
+        /* execute */
+        as(SUPER_ADMIN).cancelJob(sechubJobUUD);
+        
+        /* test */
+        waitForJobStatusCancelRequested(project, sechubJobUUD);
+        waitForJobResultFailed(project, sechubJobUUD);
+        
+        
+        /* @formatter:on */
+    }
+    
+    @Test
+    /**
+     * We start a long running job and start a cancel operation here
+     */
+    public void cancel_a_long_running_codescan_job() {
+        /* @formatter:off */
+        /* prepare */
+        UUID sechubJobUUD = as(USER_1).triggerAsyncCodeScanWithPseudoZipUpload(project,IntegrationTestMockMode.CODE_SCAN__CHECKMARX__GREEN__LONG_RUNNING);
         waitForJobRunning(project, sechubJobUUD);
         
         /* execute */

--- a/sechub-schedule/src/main/java/com/daimler/sechub/domain/schedule/ExecutionState.java
+++ b/sechub-schedule/src/main/java/com/daimler/sechub/domain/schedule/ExecutionState.java
@@ -11,7 +11,7 @@ public enum ExecutionState {
 
 	CANCEL_REQUESTED("A cancel was requested - but not ended now"),
 
-	ENDED("");
+	ENDED("Has ended - with failure or success");
 
 	private String description;
 

--- a/sechub-schedule/src/main/java/com/daimler/sechub/domain/schedule/batch/JobRegistryBeanPostProcessorFactory.java
+++ b/sechub-schedule/src/main/java/com/daimler/sechub/domain/schedule/batch/JobRegistryBeanPostProcessorFactory.java
@@ -1,0 +1,21 @@
+package com.daimler.sechub.domain.schedule.batch;
+
+import org.springframework.batch.core.configuration.JobRegistry;
+import org.springframework.batch.core.configuration.support.JobRegistryBeanPostProcessor;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.context.annotation.Bean;
+import org.springframework.stereotype.Component;
+
+@Component
+public class JobRegistryBeanPostProcessorFactory {
+
+    @Autowired
+    JobRegistry jobRegistry;
+
+    @Bean
+    public JobRegistryBeanPostProcessor jobRegistryBeanPostProcessor() {
+        JobRegistryBeanPostProcessor postProcessor = new JobRegistryBeanPostProcessor();
+        postProcessor.setJobRegistry(jobRegistry);
+        return postProcessor;
+    }
+}

